client/components/pie-chart/index.js

```diff
@@ -0,0 +1,91 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import PropTypes from 'prop-types';
+import { pie as d3Pie, arc as d3Arc } from 'd3-shape';
+
+/**
+ * Internal dependencies
+ */
+import LegendItem from './legend-item';
+
+const NUM_COLOR_SECTIONS = 3;
+const SVG_SIZE = 300;
+
+class PieChart extends Component {
+	static propTypes = {
+		data: PropTypes.array.isRequired,
+		plural: PropTypes.string,
+	};
+
+	drawChart() {
+		const { data } = this.props;
+
+		const arcs = d3Pie()
+			.startAngle( Math.PI )
+			.startAngle( -Math.PI )
+			.value( d => d.value )( data );
+		const arcGen = d3Arc()
+			.innerRadius( 0 )
+			.outerRadius( SVG_SIZE / 2 );
+
+		const paths = arcs.map( arc => arcGen( arc ) );
+
+		let sectionNum = -1;
+
+		return (
+			<g transform={ `translate(${ SVG_SIZE / 2 }, ${ SVG_SIZE / 2 })` }>
+				{ paths.map( ( path, index ) => {
+					sectionNum = ( sectionNum + 1 ) % NUM_COLOR_SECTIONS;
+					return (
+						<path
+							className={ `pie-chart__chart-section-${ sectionNum }` }
+							key={ index.toString() }
+							d={ path }
+						/>
+					);
+				} ) }
+			</g>
+		);
+	}
+
+	render() {
+		const { data, plural } = this.props;
+		// const dataSorted = data.sort((a, b) => a - b );
+		const dataTotal = data.reduce( ( pv, cv ) => pv + cv.value, 0 );
```

Comment:
What you think about moving this, `arcs` and `arcGen`
to c-tor / `getDerivedStateFromProps` / `componentWillRecieveProps` ?